Source/MatcherFactory.cs

```diff
@@ -61,11 +66,6 @@ public static IMatcher CreateMatcher(Expression expression, bool isParams)
 			// the values are ints, but if the method to call 
 			// expects, say, a double, a Convert node will be on 
 			// the expression.
-			if (isParams && (expression.NodeType == ExpressionType.NewArrayInit || !expression.Type.IsArray))
```

Comment:
This gets moved up a few lines because the preceding comment belongs with code further below; but shouldn't the type conversion code be the first thing in the method instead?